### ISSUE TITLE
Error adding new version

### ISSUE THREAD

**Issue by ZsoltPath** [Crazedpeanut/terraform-provider-luis#1]
Hi,

I'm trying to import an exported luis version to a new app. I'm getting the following response:
`Error: Could not create version [POST /apps/{appId}/versions/import/][400] importVersionJsonBadRequest  &{Error:0xc000369520}`

```
terraform version
Terraform v0.12.19
+ provider.azuread v0.7.0
+ provider.azurerm v2.0.0
+ provider.luis (unversioned)
+ provider.template v2.1.0
```

```
provider "luis" {
  key = "*****************"
}

data "template_file" "luis_json" {
  template = file("${path.module}/files/l.json")
}

resource "luis_application" "main" {
  name    = local.name
  culture = "en-us"
}

resource "luis_version" "main" {
  app_id     = luis_application.main.id
  version_id = "0.0.7"
  content    = data.template_file.luis_json.rendered
  trained    = true
  published  = true
}
```

I've tried TRACE log, but didn't make me wiser.

I was trying to upload the same file with curl and that's working fine
`curl -vXPOST -H "Content-Type: application/json" -H "Ocp-Apim-Subscription-Key: ***************" --data @l.json "https://westus.api.cognitive.microsoft.com/luis/api/v2.0/apps/b6887136-7caa-4c5d-a86d-31f1b1327fab/versions/import?versionId=0.0.6"`

Could you help me in this?

**Comment by ZsoltPath** [comment]
I forgot to add, that the luis_application is created properly. The latest terraform apply only wants to create the new version, nothing else.

**Comment by crazedpeanut** [comment]
Hey @ZsoltPath could you send through the exported version?

**Comment by crazedpeanut** [comment]
Also could you vote for an official terraform provider ? 😁

https://cognitive.uservoice.com/forums/551524-language-understanding-luis/suggestions/39795295-terraform-provider-for-authoring

**Comment by ZsoltPath** [comment]
Here you are.
Although as I said I can import this via curl command or postman.
Maybe it has something to do with the encoding. I believe it's a UTF-8 file.

[l.zip](https://github.com/Crazedpeanut/terraform-provider-luis/files/4323247/l.zip)

**Comment by crazedpeanut** [comment]
I think I have fixed it in 3670b96bff8c1db8c4a53dc419e8efd765521b2c 
I wasn't parsing the json object properly.
I think you are missing some utterances for your intents :)

**Comment by ZsoltPath** [comment]
Unfortunately I still got the same response.
I know, my original export is 14K lines long. I've just cut most of the utterances to keep it manageable size. But importing via curl still works, so that shouldn't be the issue. I guess, you've tried that.

Here is my result attached. You can see the compilation and everything.
[result.zip](https://github.com/Crazedpeanut/terraform-provider-luis/files/4328848/result.zip)

**Comment by crazedpeanut** [comment]
Hey sorry, I didn't create a new release. I assumed you cloned the repo, just creating a new one now.

**Comment by ZsoltPath** [comment]
As you can see at the top of the log I've pulled the repo up to 3670b96.

**Comment by crazedpeanut** [comment]
Sorry, missed that. You are definitely on the latest commit. What's interesting is the error message you receive. 
Error: Could not create version [POST /apps/{appId}/versions/import/][400] importVersionJsonBadRequest &{Error:0xc0006001a0}

I have changed this message so that it indicates whether it is importing a Ludown or JSON formatted version.

Your logs are missing this. Triple check the new version of the binary is being used. 

I'm not able to replicate this after the recent fixes.

**Comment by ZsoltPath** [comment]
Sorry, I haven't had time to look at this again. 
But now I've made sure that rebuild the provider from scratch. At least I think so. 
If there's a trick to make a clean build beside `go clean; go build`, let me know.

But still at the end I've got the same result.

**Comment by crazedpeanut** [comment]
Hey, i'll have another look

**Comment by crazedpeanut** [comment]
Unable to replicate this still. I have attached my logs
[out.txt](https://github.com/Crazedpeanut/terraform-provider-luis/files/4366631/out.txt)

Can you double check you have ran the following, you need to re-copy the plugin binary to the .terraform/plugins dir
```
cd ~/terraform-provider-luis
git pull
make build
cp ~/go/bin/terraform-provider-luis ~/repos/infra-path-bot/terraform.d/plugins/linux_amd64
cd ~/repos/infra-path-bot/
terraform init
```

**Comment by ZsoltPath** [comment]
Nope. Still the same. 
See my log: 
[out.txt](https://github.com/Crazedpeanut/terraform-provider-luis/files/4367972/out.txt)

**Comment by crazedpeanut** [comment]
Honestly I am stumped! For my sanity, could you try download this binary into your terraform.d dir, https://github.com/Crazedpeanut/terraform-provider-luis/releases/download/v0.0.2/terraform-provider-luis

**Comment by ZsoltPath** [comment]
F@$*&ck!
I've just realized that having a provider binary in the main directory. And apparently TF used that instead of the one in the terraform.d. 
I've tried with my compilation too and both worked nicely.

Only one thought, and it isn't a real error. And I'm not even sure TF can handle this.
When creating a new version it gives this:
```
        publish_version_direct = false
        published              = true
        trained                = true
      ~ version_id             = "0.0.7" -> "0.0.8" # forces replacement
    }
```
However I'd like to have a new version, not replacing it.
Maybe it's something MS should address.
